drivers/i2c/i2c_ite_it8xxx2.c

```diff
@@ -774,6 +774,10 @@ static int i2c_it8xxx2_transfer(const struct device *dev, struct i2c_msg *msgs,
 		/* Wait for the transfer to complete */
 		/* TODO: the timeout should be adjustable */
 		res = k_sem_take(&data->device_sync_sem, K_MSEC(100));
+		/* If the message responds with NACK, drop the transaction. */
+		if (data->err)
+			break;
```

Comment:
Update the comment to note that the transaction is dropped on any error (timeout, NACK, fail, bus error, device error).